cli/cmd/login.go

```diff
@@ -0,0 +1,183 @@
+			}
+		}
+		if err == nil && resp.StatusCode == http.StatusOK {
+			return nil
+		}
+		if delay > maxWait {
+			return fmt.Errorf("failed to connect to local server. error code: %d", resp.StatusCode)
+		}
+		time.Sleep(delay)
+		totalWait += int(delay)
+	}
+}
+
+func openBrowser(url string) error {
+	switch runtime.GOOS {
+	case "linux":
+		if err := exec.Command("xdg-open", url).Start(); err != nil {
+			return fmt.Errorf("failed to open browser with xdg-open: %w", err)
+		}
+	case "windows":
+		if err := exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Start(); err != nil {
+			return fmt.Errorf("failed to open browser with rundll32: %w", err)
+		}
+	case "darwin":
+		if err := exec.Command("open", url).Start(); err != nil {
+			return fmt.Errorf("failed to open browser with open: %w", err)
+		}
+	default:
+		return fmt.Errorf("unsupported platform for browser auto-open: %s", runtime.GOOS)
+	}
+	return nil
+}
+
+func runLogin(ctx context.Context, userConfig string) error {
+	mux := http.NewServeMux()
+	refreshToken := ""
+	gotToken := make(chan struct{})
+	mux.HandleFunc("/callback", func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, accountsURL+"/success-close", http.StatusSeeOther)
+		refreshToken = r.URL.Query().Get("token")
+		close(gotToken)
+	})
+	mux.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprintf(w, "OK")
+	})
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return fmt.Errorf("failed to listen: %w", err)
+	}
+	server := http.Server{
+		Handler: mux,
+		Addr:    listener.Addr().String(),
+	}
+
+	var serverErr error
+	go func() {
+		if err := server.Serve(listener); err != nil {
+			if err != http.ErrServerClosed {
+				serverErr = fmt.Errorf("failed to serve: %w", err)
+			}
+		}
+	}()
+	localServerAddr := fmt.Sprintf("http://localhost:%d", listener.Addr().(*net.TCPAddr).Port)
+	if err := waitForServer(ctx, localServerAddr); err != nil {
+		return err
+	}
+	url := fmt.Sprintf("%s?returnTo=%s/callback", accountsURL, localServerAddr)
+	if err := openBrowser(url); err != nil {
+		fmt.Printf("Failed to open browser at %s. Please open the URL manually.\n", accountsURL)
+	} else {
+		fmt.Printf("Opened browser at %s. Waiting for authentication to complete.\n", url)
+	}
+
+	// Wait for an OS signal to begin shutting down.
+	select {
+	case <-ctx.Done():
+		fmt.Println("Context cancelled. Shutting down server.")
+	case <-gotToken:
+	}
+
+	// Create a context for the shutdown with a 5-second timeout.
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
+	defer cancel()
+	if err := server.Shutdown(ctx); err != nil {
+		return fmt.Errorf("failed to shutdown server: %w", err)
+	}
+
+	if serverErr != nil {
+		return serverErr
+	}
+
+	if refreshToken == "" {
+		return fmt.Errorf("failed to get refresh token")
+	}
+	tokenPath := path.Join(userConfig, "token")
```

Comment:
What is the format of the token file?
Maybe it use be a JSON with the following format:

`~/.config/cloudquery/auth.json`

```js
{
  "version": 1,
  "token": ...
}
```

(We could even have a list of tokens for multiple accounts, but probably best to do it when we need it)